server/graphql/v2/interface/Account.js

```diff
@@ -393,6 +393,29 @@ const accountFieldsDefinition = () => ({
   location: {
     type: Location,
     description: 'The address associated to this account. This field is always public for collectives and events.',
+    async resolve(collective, _, req) {
+      const canSeeLocation =
+        collective.type !== CollectiveTypes.USER ||
+        (req.remoteUser?.isAdmin(collective.id) && checkScope(req, 'account')) ||
+        (await collective.isHost());
+
+      if (!canSeeLocation) {
+        return null;
+      }
+
+      // For incognito profiles, we retrieve the location from the main user profile
+      if (collective.isIncognito) {
+        if (!checkScope(req, 'incognito')) {
+          return null;
+        }
+        const mainProfile = await req.loaders.Collective.mainProfileFromIncognito.load(collective.id);
+        if (mainProfile) {
+          return mainProfile.getLocation();
+        }
+      }
+
+      return collective.getLocation();
+    },
```

Comment:
This is moved from `../graphql/v2/object/Individual.js` to easier be able to see permissions for the `location` field in one place.

Worth noting is that there currently is a disrepancy between GQL v1 and v2 for what Collective types they expose location, v1 exposes for `COLLECTIVE`, `EVENT`, `ORGANIZATION`, and v2 for all types except `USERS`.